src/plugins/intel_cpu/src/transformation_pipeline.cpp

```diff
@@ -0,0 +1,603 @@
+#include <cpu/x64/cpu_isa_traits.hpp>
+
+namespace ov {
+namespace intel_cpu {
+
+using const_node_ptr = const std::shared_ptr<const ov::Node>;
+
+bool Transformations::fuse_type_to_convert(const std::shared_ptr<ngraph::Node>& node, ov::element::Type to, size_t idx) {
+    if (auto convert = ov::as_type_ptr<ov::opset10::Convert>(node)) {
+        // For Convert node, converting precision from floating point to boolean will lead to mathematical
+        // error, because here the output precision boolean is replaced by u8. E.g. floating point value 0.01
+        // is converted to be 1 for boolean, but 0 for u8. Thus an Abs and Ceil node should be added before the
+        // Convert node for this scenario.
+        if (convert->input(0).get_element_type().is_real() &&
+            convert->get_convert_element_type() == ngraph::element::boolean && to.is_integral_number()) {
+            auto abs = std::make_shared<ov::opset10::Abs>(convert->input_value(0).get_node_shared_ptr());
+            auto ceil = std::make_shared<ov::opset10::Ceiling>(abs);
+            auto new_convert = std::make_shared<ov::opset10::Convert>(ceil, to);
+            new_convert->set_friendly_name(convert->get_friendly_name());
+            ov::copy_runtime_info(convert, {abs, ceil, new_convert});
+            ov::replace_node(convert, new_convert);
+            return true;
+        } else {
+            convert->set_convert_element_type(to);
+            return true;
+        }
+    }
+    return false;
+}
+
+void Transformations::upToCpuSpecificOpSet() {
+    const bool useLpt = enableLpt &&
+        ngraph::pass::low_precision::LowPrecision::isFunctionQuantized(model) &&
+        CPU_DEBUG_CAP_IS_TRANSFORMATION_ENABLED(config, Lpt);
+
+    const bool useSnippets = enableSnippets &&
+        CPU_DEBUG_CAP_IS_TRANSFORMATION_ENABLED(config, Snippets);
+
+    auto defaultPrecisions = useLpt ? ngraph::pass::low_precision::precision_set::int8_support : std::vector<ov::element::Type>{};
+    bool hasINT16orINT32Levels = false;
+
+    if (useLpt) {
+        CPU_LPT_SCOPE(LowPrecisionTransformations_Part1);
+        hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
+            model,
+            {ngraph::pass::low_precision::levels::int16, ngraph::pass::low_precision::levels::int16_narrow_range,
+             ngraph::pass::low_precision::levels::int32, ngraph::pass::low_precision::levels::int32_narrow_range});
+        if (hasINT16orINT32Levels) {
+            defaultPrecisions = ngraph::pass::low_precision::precision_set::int8_int16_int32_support;
+        }
+    }
+
+    upToCpuSpecificOpSet_preLpt(defaultPrecisions, isLegacyApi);
+
+    if (useLpt)
+        upToCpuSpecificOpSet_lpt(hasINT16orINT32Levels, defaultPrecisions);
+
+    upToCpuSpecificOpSet_postLpt();
+
+    if (useSnippets)
+        upToCpuSpecificOpSet_snippets();
+}
+
+void Transformations::cpuSpecificOpSet(void) {
+    CPU_DEBUG_CAP_TRANSFORMATION_DUMP(this, Specific);
```

Comment:
Is there any way how I can disable cpuSpecificOpSet transformations via env variable?
I was thinking that we can inttroduce some generic CPU_TRANSFORMATION_SCOPE macro (placed in the beggining of specific transformation method) that will expand in several other macroses underneath: 1) check id tranformation is enabled and return if not, 2) Creates TransformationDumper 3) Inserts ITT conunter ... etc
That should help to encapsulate all debug/auxiliary related functionality into single place and minimize the chance of braking smt during future develompent.

Reply:
this is exactly how it is done in this PR, just wrong macro is used for some reason
Corrected

Reply:
Regarding replacing _RETURN_OR_DUMP with _SCOPE macro and adding ITT scope into every transformation step, I propose to do it in scope of separate PR

Reply:
Lets rename to CPU_TRANSFORMATION_SCOPE in bounds of this PR.
ITT counters can be processed as separate task.

Reply:
Done